OneCommunityGlobal/HGNRest
# Sav fix lead team of x badge

# Description
This bug resulted from improper backend calls and a typo in the aggregation pipeline of the database for the myTeam collection and null values in backend permissions of several roles. I have removed permissions from the function so it will run faster and refactored the function. This badge should be awarded to users with either the mentor or the manager role and will only count users without the roles of Admin, Owner, Coreteam, manager, or mentor as team members being led. For example, if your user was on a team with 10 users with leadership roles and 5 volunteers they would be awarded a badge for leading a team of 5 users. To test this function make a user a mentor or manager and then add them to a team of at least 5 volunteers to make sure this badge is being awarded properly. Best of luck please watch the video for more clarification.

## Main changes explained:
Fix getTeamMembers function to call database properly
Remove hasPermission calls.
Refactor code to run faster.
…

## How to test:
check into current branch
log as admin user
Make a mentor/manager user
Add user to team with 5 or more volunteers 
isolate tested mentor/manager user in awardBadges
## Screenshots or videos of changes:
https://github.com/OneCommunityGlobal/HGNRest/assets/98275089/9bb0693a-948b-47af-9815-c1dae67d2087
